File: output/ChatGPT/avatar/Java/C++/atcoder_ABC145_B.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string a = "", b = "", str;
    int n;
    cin >> n;
    cin.ignore();
    getline(cin, str);
    for (int i = 0; i < n / 2; i++) {
        a += str[i];
    }
    if (str == a + a) {
        cout << "Yes" << endl;
    } else {
        cout << "No" << endl;
    }
}

File: output/CodeLlama/avatar/Java/Python/atcoder_ABC145_B.py
import sys
import math

def cout(str):
    print(str)

if __name__ == '__main__':
    cin = sys.stdin
    n = int(cin.readline())
    str = cin.readline()
    a = ""
    b = ""
    for i in range(n // 2):
        a += str[i]
    if str == a + a:
        cout("Yes")
    else:
        cout("No")